Link without -rdynamic.

GCCSDK's linker doesn't support -rdynamic, which OpenTTD only uses for
nicer symbol names in crash reports.

For OpenTTD 14.1. See docs/PORTING-NOTES.md in riscos-openttd.

diff --git a/cmake/CompileFlags.cmake b/cmake/CompileFlags.cmake
index a85a915..88af179 100644
--- a/cmake/CompileFlags.cmake
+++ b/cmake/CompileFlags.cmake
@@ -149,7 +149,7 @@ macro(compile_flags)
         message(FATAL_ERROR "No warning flags are set for this compiler yet; please consider creating a Pull Request to add support for this compiler.")
     endif()
 
-    if(NOT WIN32 AND NOT HAIKU)
+    if(NOT WIN32 AND NOT HAIKU AND NOT RISCOS)
         # rdynamic is used to get useful stack traces from crash reports.
         set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -rdynamic")
     endif()
